File: JavaScript_12.18/1234 - Sentença dançante.js
/**
 * @author Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
 * @copyright Josélio Júnior (Lunatic Fox) 2022
*/

const l = require('fs').readFileSync('/dev/stdin', 'utf8').split('\n');
while (true) {
  let get = l.shift();
  if (!get) break;
  let k = true;
  get = get.split('')
    .map(e => {
      if (e.match(/\w/)) {
        if (k) {
          e = e.toUpperCase();
          k = false;
        } else {
          e = e.toLowerCase();
          k = true;
        }
      }
      return e;
    })
    .join('');
  console.log(get);
}

File: Lua_5.4/1234 - Sentença Dançante.lua
--- author: Josélio de S. C. Júnior <joseliojrx25@gmail.com>
--- copyright: Josélio de S. C. Júnior 2022

while true do
  local t, k, r = {}, true, io.read()
  if not r then break end
  for e in r:gmatch '.' do
    if e:match '%a' then
      if k then
        table.insert(t, e:upper())
        k = false
      else
        table.insert(t, e:lower())
        k = true
      end
    else
      table.insert(t, e)
    end
  end
  print(table.concat(t, ''))
end
